h/views/api_auth.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from datetime import (datetime, timedelta)
+
+import jwt
+
+from h import models
+from h.util.view import json_view
+
+
+class OAuthTokenError(Exception):
+    def __init__(self, message, description=None, status_code=400):
+        self.description = description
+        self.status_code = status_code
+        super(OAuthTokenError, self).__init__(message)
+
+
+@json_view(route_name='token', request_method='POST')
+def access_token(request):
+    grant_token = _validate_request(request)
+    userid = _extract_userid(request, grant_token)
+
+    user = request.db.query(models.User).filter_by(userid=userid).one_or_none()
+    if user is None:
+        raise OAuthTokenError('invalid_grant',
+                              description='user with userid described in subject could not be found')
+
+    expires_in = timedelta(hours=1)
+    token = models.Token(userid=user.userid, expires=(utcnow() + expires_in))
+    request.db.add(token)
+
+    return {
+        'access_token': token.value,
+        'token_type': 'bearer',
+        'expires_in': expires_in.seconds,
+    }
+
+
+@json_view(context=OAuthTokenError)
+def api_token_error(context, request):
+    """Handle an expected/deliberately thrown API exception."""
+    request.response.status_code = context.status_code
+    resp = {'error': context.message}
+    if context.description:
+        resp['error_description'] = context.description
```

Comment:
Sean is right (RFC6749, OAuth, to be exact).
https://tools.ietf.org/html/rfc6749#appendix-A.8